internal/integration/testcase_test.go

```diff
@@ -89,9 +90,11 @@ func (t testcase) CreatePipeline(ctx context.Context) (string, func()) {
 	})
 	require.NoError(t, err)
 
-	return *pipeline.GraphQLID, func() {
-		if !preservePipelines {
-			t.deletePipeline(ctx)
+	return *pipeline.GraphQLID, func(t testcase) func() {
```

Comment:
Why does the cleanup closure need to take the (same) `testcase` as an arg? afaict a copy of `testcase` will contain the same `*testing.T` and `*buildkite.Client`, so the cleanup closure should work on the captured `t` without having to pass it back in.

Reply:
I don't know what I was thinking when I wrote that. I was probably exhausted after many test failures that I could not debug!

I've simplified it significantly now.